.github/workflows/openliberty-ci.yml

```diff
@@ -126,41 +125,40 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-maven-
     - name: Run unit tests
-      run: |
-        cd dev
-        chmod +x gradlew
-        ./gradlew --continue cnf:initialize testResults
-    - name: Upload test report
-      if: failure()
+      id: run-unit-tests
+      run: .github/workflow-scripts/run-unit-tests.sh
+    - name: Upload unit test artifact
+      if: steps.run-unit-tests.outputs.status == 'failure'
       uses: actions/upload-artifact@v2
       with:
-        name: Unit test logs
+        name: Unit Test Logs
         path: dev/cnf/generated/testReports/allUnitTests
-    - name: Upload unit test results
-      shell: bash
-      if: always()
-      run: .github/workflow-scripts/upload-unit-tests.sh
-  fat_tests:
+    - name: Analyze junit output
+      if: steps.run-unit-tests.outputs.status == 'failure'
+      uses: KyleAure/junit-report-annotations-action@1.6
+      with:
+        name: Unit test report
+        path: "dev/unit-results/TEST-*.xml"
+        includeSummary: true
+        numFailures: 10 # Truncate number of failures so we don't run out of annoations.
```

Comment:
what's the reasoning behind limiting num failures? Do we have a limited amount of annotations on each build? Or is it just so we avoid flooding logs/results with failures?